plugins/tangoinfo/tangoinfo.py

```diff
@@ -0,0 +1,360 @@
+                          ("a zero" if zeros == 1 else "%d zeros" % zeros))
+
+            for track, album, tint in track_triple:
+                tm = track.metadata
+                if not self.albumpage_cache[barcode].get(tint):
+                    self.album_remove_request(album)
+                    continue
+
+                for field in ["genre", "date", "vocal"]:
+                    # Write track metadata
+                    if self.albumpage_cache[barcode][tint].get(field):
+                        tm[field] = self.albumpage_cache[barcode][tint][field]
+
+                for file in track.iterfiles():
+                    fm = file.metadata
+                    for field in ["genre", "date", "vocal"]:
+                        if not self.albumpage_cache[barcode][tint].get(field):
+                            continue
+                        # Write file metadata
+                        fm[field] = self.albumpage_cache[barcode][tint][field]
+                self.album_remove_request(album)
+        else:
+            if zeros >= 2:
+                log.debug("%s: Could not load album with barcode %s even with "
+                          "zero prepended(%s). This most likely means "
+                          "tango.info does not have a release for this "
+                          "barcode (or MusicBrainz has a wrong barcode)",
+                          PLUGIN_NAME, barcode, ("0" * zeros) + barcode)
+                for track, album, tint in track_triple:
+                    self.album_remove_request(album)
+                return
+
+            log.debug("%s: Retrying with 0-padded barcode for barcode %s",
+                      PLUGIN_NAME, barcode)
+
+            for track, album, tint in track_triple:
+                # Try again with zero-prepended barcode, but at most two times
+                self.website_add_track(
+                    album, track, barcode, tint, zeros=(zeros + 1)
+                )
+                self.album_remove_request(album)
+
+    def album_add_request(self, album):
+        album._requests += 1
+
+    def album_remove_request(self, album):
+        album._requests -= 1
+        album._finalize_loading(None)
+
+    def extract_data(self, barcode, response):
+
+        # Check whether we have a concealed 404 and get the homepage
+        if "<title>Contents - tango.info</title>" in response:
+            log.debug("%s: No album with barcode %s on tango.info",
+                      PLUGIN_NAME, barcode)
+            return
+
+        table = re.findall(table_regex, response)
+        if not table:
+            log.warning("%s: Could not extract table from album webpage - "
+                        "regex failed or page structure changed", PLUGIN_NAME)
+            return
+        table = table[0]  # re.findall() returns a list
+
+        # Content inside of <tr> elements
+        trcontent = [match.groups()[0] for match in tr_regex.finditer(table)]
```

Comment:
This could be made a generator:

```suggestion
        trcontent = (match.groups()[0] for match in tr_regex.finditer(table))
```

Reply:
Done